shell/platform/common/cpp/flutter_platform_node_delegate.h

```diff
@@ -0,0 +1,69 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PLATFORM_COMMON_CPP_FLUTTER_PLATFORM_NODE_DELEGATE_H_
+#define FLUTTER_SHELL_PLATFORM_COMMON_CPP_FLUTTER_PLATFORM_NODE_DELEGATE_H_
+
+#include "flutter/shell/platform/embedder/embedder.h"
+
+#include "flutter/third_party/accessibility/ax/ax_event_generator.h"
+#include "flutter/third_party/accessibility/ax/platform/ax_platform_node_delegate_base.h"
+
+namespace flutter {
+
+class AccessibilityBridge;
+
+//------------------------------------------------------------------------------
+/// The accessibility node delegate to be used in accessibility bridge. This
+/// class is responsible for providing native accessibility object with
+/// appropriate information, such as accessibility label/value/bounds.
+///
+/// While most methods have default implementations and are ready to be used
+/// as-is, the subclasses must override the GetNativeViewAccessible to return
+/// native accessibility objects. To do that, subclasses should create and
+/// maintain AXPlatformNode[s] which delegate their accessibility attributes to
+/// this class.
+///
+/// For desktop platforms, subclasses also need to override the GetBoundsRect
+/// to apply window-to-screen transform.
+class FlutterPlatformNodeDelegate : public ui::AXPlatformNodeDelegateBase {
+ public:
+  FlutterPlatformNodeDelegate();
+  ~FlutterPlatformNodeDelegate() override;
+
+  //------------------------------------------------------------------------------
+  /// @brief      Gets the accessibility bridge to which this accessibility node
+  ///             belongs.
+  AccessibilityBridge* GetBridge() const;
```

Comment:
This shouldn't be a raw pointer.

Reply:
I removed this getter because subclass should figure out their way to access the bridge in the embedding. but it still need to be stored in this class as a raw pointer, I couldn't figure out a way to use smart pointer in this case without forcing the owner of accessiblity bridge to hold it as shared_ptr.